internal/usecases/get_cluster_credentials.go

```diff
@@ -129,6 +123,49 @@ func (u *getClusterCredentialsUseCase) run(ctx context.Context) error {
 	return nil
 }
 
+// getAllClustersAuthInformation gets a token per cluster to mimik cross cluster login (for now)
+func (u *getClusterCredentialsUseCase) getAllClustersAuthInformation(ctx context.Context) {
+	wg := &sync.WaitGroup{}
+	m8Clusters, _ := u.clusterServiceClient.GetAll(ctx, &api.GetAllRequest{IncludeDeleted: false})
+	for {
+		m8Cluster, err := m8Clusters.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			continue
+		}
+
+		clusterAuthInfo := u.config.GetClusterAuthInformation(m8Cluster.Name, u.config.AuthInformation.Username, u.clusterRole)
+		if clusterAuthInfo != nil && clusterAuthInfo.IsValidExact() {
+			continue
+		}
+
+		go func() {
+			wg.Add(1)
+			defer wg.Done()
+			_, _ = u.requestClusterAuthInformation(ctx, m8Cluster)
```

Comment:
What if this fails in the go routine? The user will not notice and be not logged in.

Reply:
This was actually intentional and my idea behind it was to really hide caching the tokens of the other clusters from the user and have it as a background job that shouldn't interact with the user to ensure he is not distracted with fixing errors of other clusters, that he might not even use in his current session.

This is why I made sure to run the first/main request (see [here](https://github.com/finleap-connect/monoctl/pull/70/files/13b668a353f57f084df289a81bf1b27f12fdf3e0#diff-f257cb054b1554ccbdbccac4a0c983c5f8f24f1c0566681d285978e909efd1e6R82)) in the main routine and forward the error back to the user and if everything goes well the probability of an error happing with the other clusters is really low. Besides that, the user may not even need to switch context for his current work so why should he deal with the error immediately.

If there is an error the user will be notified as soon as he switches to the cluster where the error was because there won't be any token cached and the main routine will ask the gateway again and return the error back to the user. Therefore I would suggest sticking with this approach.